src/Components/DateInputForm/DateInputForm.tsx

```diff
@@ -1,22 +1,37 @@
-import React from "react";
+import React, {FormEvent, SetStateAction, useState} from "react";
 import "./DateInputForm.scss"
+import {getRoverImages} from "../ApiFetch/RoversApiFetch";
+import {RoverGalleryImage} from "../RoverGalleryImage/RoverGalleryImage";
 
 
-export function DateInputForm() {
+interface DateInputFormProps {
+    dateInput : string | null;
+    setDateInput : (date: string) => void;
+}
+
+export function DateInputForm(props : DateInputFormProps) {
+    const [date, setDate] = useState("");
+    
     function getDateToday(){
         let today: string = new Date().toISOString().split("T")[0];
         return today;
     }
+    
+    function handleClick(event: FormEvent) {
+        console.log(date);
+        event.preventDefault();
+        props.setDateInput(date);
+    }
 
     return (
         <div >
-            <form className="form">
+            <form data-testid={"on-this-day-form"} onSubmit={handleClick} className="form">
                 <label>DATE:
                     <br/>
-                    <input type="date" id="date" name="date"
+                    <input data-testid={"date-input"} value={date} onChange={event => setDate(event.target!.value)} type="date" id="date" name="date"
```

Comment:
What does the !. do?

Reply:
https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-guards-and-type-assertions

Basically it tells TypeScript `I promise this this isn't null or undefined` and then lets you continue as if its no longer optional.

Reply:
Maaaagic!

Reply:
Neat!